flink-python/pyflink/fn_execution/coders.py

```
@@ -271,6 +309,21 @@ def __init__(self, precision, scale):
     def get_impl(self):
         return coder_impl.DecimalCoderImpl(self.precision, self.scale)
 
+    def get_slow_impl(self):
+        return coder_impl_slow.DecimalCoderImpl(self.precision, self.scale)
+
+
+class BasicDecimalCoder(FieldCoder):
+    """
+    Coder for Basic Decimal that no need to have precision and scale specified.
+    """
+
+    def get_impl(self):
+        pass
```

I think return None is not a good chocie. Maybe you can raise a exception